File: app/src/main/java/wooyun/esnb/dialog/Popupwindow.java
package wooyun.esnb.dialog;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Context;
import android.graphics.drawable.ColorDrawable;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.PopupWindow;
import android.widget.TextView;

import wooyun.esnb.R;

public class Popupwindow extends PopupWindow {
    private View mMenuView;

    @SuppressLint({"InflateParams", "ClickableViewAccessibility"})
    public Popupwindow(Activity context, View.OnClickListener itemsOnClick) {
        super(context);
        LayoutInflater inflater = (LayoutInflater) context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        mMenuView = inflater.inflate(R.layout.alert_show, null);
        TextView btn_bianhua = (TextView) mMenuView.findViewById(R.id.btn_bianhua);
        TextView btn_dchu = (TextView) mMenuView.findViewById(R.id.btn_dchu);

        //设置按钮监听
        btn_dchu.setOnClickListener(itemsOnClick);
        btn_bianhua.setOnClickListener(itemsOnClick);
        //设置SelectPicPopupWindow的View
        this.setContentView(mMenuView);
        //设置SelectPicPopupWindow弹出窗体的宽
        this.setWidth(ViewGroup.LayoutParams.FILL_PARENT);
        //设置SelectPicPopupWindow弹出窗体的高
        this.setHeight(ViewGroup.LayoutParams.WRAP_CONTENT);
        //设置SelectPicPopupWindow弹出窗体可点击
        this.setFocusable(true);
        //设置SelectPicPopupWindow弹出窗体动画效果
        this.setAnimationStyle(R.style.AnimBottom);
        //实例化一个ColorDrawable颜色为半透明
        ColorDrawable dw = new ColorDrawable(0xb0000000);
        //设置SelectPicPopupWindow弹出窗体的背景
        this.setBackgroundDrawable(dw);
        //mMenuView添加OnTouchListener监听判断获取触屏位置如果在选择框外面则销毁弹出框
        mMenuView.setOnTouchListener((v, event) -> {

            int height = mMenuView.findViewById(R.id.pop_layout).getTop();
            int y=(int) event.getY();
            if(event.getAction()==MotionEvent.ACTION_UP){
                if(y<height){
                    dismiss();
                }
            }
            return true;
        });

    }

}

File: app/src/main/java/wooyun/esnb/dialog/Popupwindow.kt
package wooyun.esnb.dialog

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Context
import android.graphics.drawable.ColorDrawable
import android.view.LayoutInflater
import android.view.MotionEvent
import android.view.View
import android.view.ViewGroup
import android.widget.PopupWindow
import android.widget.TextView
import wooyun.esnb.R

class Popupwindow @SuppressLint("InflateParams", "ClickableViewAccessibility") constructor(context: Activity, itemsOnClick: View.OnClickListener?) : PopupWindow(context) {
    private val mMenuView: View

    init {
        val inflater = context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        mMenuView = inflater.inflate(R.layout.alert_show, null)
        val btn_bianhua = mMenuView.findViewById<View>(R.id.btn_bianhua) as TextView
        val btn_dchu = mMenuView.findViewById<View>(R.id.btn_dchu) as TextView

        //设置按钮监听
        btn_dchu.setOnClickListener(itemsOnClick)
        btn_bianhua.setOnClickListener(itemsOnClick)
        //设置SelectPicPopupWindow的View
        this.contentView = mMenuView
        //设置SelectPicPopupWindow弹出窗体的宽
        this.width = ViewGroup.LayoutParams.FILL_PARENT
        //设置SelectPicPopupWindow弹出窗体的高
        this.height = ViewGroup.LayoutParams.WRAP_CONTENT
        //设置SelectPicPopupWindow弹出窗体可点击
        this.isFocusable = true
        //设置SelectPicPopupWindow弹出窗体动画效果
        this.animationStyle = R.style.AnimBottom
        //实例化一个ColorDrawable颜色为半透明
        val dw = ColorDrawable(-0x50000000)
        //设置SelectPicPopupWindow弹出窗体的背景
        setBackgroundDrawable(dw)
        //mMenuView添加OnTouchListener监听判断获取触屏位置如果在选择框外面则销毁弹出框
        mMenuView.setOnTouchListener { v: View?, event: MotionEvent ->
            val height = mMenuView.findViewById<View>(R.id.pop_layout).top
            val y = event.y.toInt()
            if (event.action == MotionEvent.ACTION_UP) {
                if (y < height) {
                    dismiss()
                }
            }
            true
        }
    }
}

